File: StructuralPatterns/Bridge/TypeScript/RasterRenderer.ts
import { Renderer } from './Renderer';

export class RasterRenderer implements Renderer {
    renderCircle(radius: number): void {
        console.log(`Drawing a circle of radius ${radius} with raster renderer.`);
    }
    renderSquare(size: number): void {
        console.log(`Drawing a square of size ${size} with raster renderer.`);
    }
}

File: StructuralPatterns/Bridge/CSharp/RasterRenderer.cs
using System;

namespace Bridge {
    public class RasterRenderer : IRenderer {
        public void RenderCircle(double radius) {
            Console.WriteLine($"Drawing a circle of radius {radius} with raster renderer.");
        }
        public void RenderSquare(double size) {
            Console.WriteLine($"Drawing a square of size {size} with raster renderer.");
        }
    }
}

